internal/server/data/group.go

```diff
@@ -41,19 +40,37 @@ func ListGroups(db GormTxn, p *Pagination, selectors ...SelectorFunc) ([]models.
 	}
 
 	return groups, nil
-
 }
 
 func ByGroupMember(id uid.ID) SelectorFunc {
 	return func(db *gorm.DB) *gorm.DB {
 		return db.
-			Joins("JOIN identities_groups ON groups.id = identities_groups.group_id").
-			Where("identities_groups.identity_id = ?", id)
+			Joins(`
+			JOIN (
+				-- Combine the direct group members and the members of mapped identity provider groups
+				SELECT * FROM identities_groups
+				NATURAL FULL JOIN (
+					-- See identities in groups based on their provider group mappings
+					SELECT identity_id, group_id from provider_groups_identities
+					JOIN provider_groups_groups ON provider_groups_groups.provider_group_id = provider_groups_identities.provider_group_id
+				) AS normalized_identity_groups
+				) AS resolved_identity_groups
+			ON resolved_identity_groups.group_id = id
+			`).
+			Where("resolved_identity_groups.identity_id = ?", id)
```

Comment:
I think we've taken a wrong turn here. this should never query anything other than identities_groups, as that's the canonical record of group membership.

Reply:
Fixed with the Postgres view `resolved_identity_groups` is not our canonical record of group membership